bin/exec.ml

```diff
@@ -111,19 +111,34 @@ let build_prog ~no_rebuild ~prog p =
     p
 ;;
 
-let get_path_and_build_if_necessary sctx ~no_rebuild ~dir ~prog =
+let dir_of_context common context =
   let open Memo.O in
+  let+ sctx = Super_context.find_exn context in
+  let context = Dune_rules.Super_context.context sctx in
+  Path.Build.relative (Context.build_dir context) (Common.prefix_target common "")
+;;
+
+let expand common context prog =
+  let open Memo.O in
+  let* sctx = Super_context.find_exn context in
```

Comment:
This function is called in 4 places or so, maybe it would make more sense to do it in one place and pass `sctx` as argument instead? `exec.ml:215` already calculates it, so might as well pass it on.

Reply:
Good point. Updated.